source/common/upstream/load_balancer_config.h

```diff
@@ -0,0 +1,336 @@
+class TypedRoundRobinLbConfig : public LoadBalancerConfig {
+public:
+  TypedRoundRobinLbConfig(
+      const envoy::extensions::load_balancing_policies::round_robin::v3::RoundRobin& lb_config);
+
+  const envoy::extensions::load_balancing_policies::round_robin::v3::RoundRobin lb_config_;
+};
+
+/**
+ * Load balancer config that used to wrap the legacy least request config.
+ */
+class LegacyTypedLeastRequestLbConfig : public LoadBalancerConfig {
+public:
+  LegacyTypedLeastRequestLbConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  absl::optional<envoy::config::cluster::v3::Cluster::LeastRequestLbConfig> lb_config_;
+};
+
+/**
+ * Load balancer config that used to wrap the least request config.
+ */
+class TypedLeastRequestLbConfig : public LoadBalancerConfig {
+public:
+  TypedLeastRequestLbConfig(
+      const envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest& lb_config);
+
+  const envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest lb_config_;
+};
+
+/**
+ * Load balancer config that used to wrap the legacy random config.
+ */
+class LegacyTypedRandomLbConfig : public LoadBalancerConfig {};
+
+/**
+ * Load balancer config that used to wrap the random config.
+ */
+class TypedRandomLbConfig : public LoadBalancerConfig {
+public:
+  TypedRandomLbConfig(
+      const envoy::extensions::load_balancing_policies::random::v3::Random& lb_config);
+
+  const envoy::extensions::load_balancing_policies::random::v3::Random lb_config_;
+};
+
+/**
+ * Load balancer config that used to wrap legacy maglev config.
+ */
+class LegacyTypedMaglevLbConfig : public Upstream::LoadBalancerConfig {
+public:
+  LegacyTypedMaglevLbConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  absl::optional<envoy::config::cluster::v3::Cluster::MaglevLbConfig> lb_config_;
+};
+
+/**
+ * Load balancer config that used to wrap typed maglev config.
+ */
+class TypedMaglevLbConfig : public Upstream::LoadBalancerConfig {
+public:
+  TypedMaglevLbConfig(const envoy::extensions::load_balancing_policies::maglev::v3::Maglev& config);
+
+  const envoy::extensions::load_balancing_policies::maglev::v3::Maglev lb_config_;
+};
+
+/**
+ * Load balancer config that used to wrap legacy ring hash config.
+ */
+class LegacyTypedRingHashLbConfig : public Upstream::LoadBalancerConfig {
+public:
+  LegacyTypedRingHashLbConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  absl::optional<envoy::config::cluster::v3::Cluster::RingHashLbConfig> lb_config_;
+};
+
+/**
+ * Load balancer config that used to wrap typed ring hash config.
+ */
+class TypedRingHashLbConfig : public Upstream::LoadBalancerConfig {
+public:
+  TypedRingHashLbConfig(
+      const envoy::extensions::load_balancing_policies::ring_hash::v3::RingHash& config);
+
+  const envoy::extensions::load_balancing_policies::ring_hash::v3::RingHash lb_config_;
+};
+
+/**
+ * Implementation of SubsetSelector
+ */
+class SubsetSelectorImpl : public SubsetSelector {
```

Comment:
are these just pure moves?
Also can we get better comments?

Reply:
Yeah, this is pure move. Will try to update the comment.